Vectorize: Make index naming requirement explicit (#22335)
cloudflare/cloudflare-docs: src/content/docs/vectorize/best-practices/create-indexes.mdx

Creating an index requires three inputs:

> - A kebab-cased name, for example `prod-search-index` or `recommendations-idx-dev`.

Reviewer: ```suggestion
- A kebab-cased name, such as `prod-search-index` or `recommendations-idx-dev`.
```
